app/src/main/java/org/mozilla/focus/utils/WifiConnectionMonitor.kt

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.focus.utils
+
+import android.app.Application
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import androidx.annotation.VisibleForTesting
+
+/**
+ * Attaches itself to the [Application] and listens for WIFI available/not available events. This
+ * allows calling code to set simpler listeners.
+ *
+ * Example:
+ * ```kotlin
+ *  app.components.wifiConnectionListener.addOnWifiConnectedChangedListener { isConnected ->
+ *    if (isConnected) {
+ *      downloadThing()
+ *    }
+ *  }
+ *  app.components.wifiConnectionListener.start()
+ * ```
+ */
+class WifiConnectionMonitor(app: Application) {
```

Comment:
@Amejia481 One of the autoplay options is "Block audio and video on cellular data only" and I added this wifi connection monitor to  address that scenario when the state of wifi network is changing during the browsing.
Since "Block audio and video on cellular data only" seems more orientated to optimization and not to privacy should we remove this option from the list?

Reply:
Yes, I think the basic options should be enough in Focus, for now. If in the future product made the decision of adding this functionality we will need to uplift it to AC.